protostar/commands/deploy/deploy_command.py

```diff
@@ -40,10 +48,47 @@ def arguments(self) -> List[Command.Argument]:
                 description="Used for deploying contracts in Alpha MainNet.",
                 type="str",
             ),
+            Command.Argument(
+                name="salt",
+                description=(
+                    "An optional salt controlling where the contract will be deployed. "
+                    "The contract deployment address is determined by the hash "
+                    "of contract, salt and caller. "
+                    "If the salt is not supplied, the contract will be deployed with a random salt."
+                ),
+                type="str",
+            ),
         ]
 
-    async def run(self, _args):
-        return await self.deploy()
+    async def run(self, args):
+        return await self.deploy(
+            contract_name=args.contract,
+            network=args.network,
+            inputs=args.inputs,
+            token=args.token,
+            salt=args.salt,
+        )
+
+    # pylint: disable=too-many-arguments
+    async def deploy(
+        self,
+        contract_name: str,
+        network: str,
+        inputs: Optional[List[str]] = None,
+        token: Optional[str] = None,
+        salt: Optional[str] = None,
+    ):
+        with open(
+            self._build_output_path / f"{contract_name}.json",
```

Comment:
This is an open question but should user manually prebuild a contract before deployment?

Reply:
Yes. I added suggestion to the user to run `build` command in case such file doesn't exist. Custom commands feature would allow the user to combine `build` and `deploy` into one command.

Reply:
The suggestion was added in this PR — https://github.com/software-mansion/protostar/pull/288.